plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/editors/groovy/GroovyCompletionProcessor.java

```diff
@@ -0,0 +1,217 @@
+    * @param viewer the editor view
+    * @param offset the cursor position in the document
+    * @return the list of the completion proposals
+    */
+	@Override
+	public ICompletionProposal[] computeCompletionProposals(ITextViewer viewer, int offset) {
+		IDocument document = viewer.getDocument();
+		int currOffset = offset-1;
+		String currWord = "";
+		if (currOffset >= 0) {
+		    try {
+				char currChar;
+				/*
+				 * Find the start position of the word where we trigger the completion.
+				 * We move from offset to offset until we meet a space or \n character.
+				 * Each character is added at the start of the current word.
+				 */
+				while (currOffset >= 0 && !Character.isWhitespace(currChar = document.getChar(currOffset))) {
+				    currWord = currChar + currWord;
+				    System.out.println("currWord : " + currWord);
+				    currOffset--;
+				}
+				
+				String LANGUAGE_GROOVY = "groovy";
+		        Path workspaceRoot = Paths.get(System.getProperty("user.dir")).resolve("./build/test_workspace/");
+		    	if (!Files.exists(workspaceRoot)) {
+		    		workspaceRoot.toFile().mkdirs();
+				}
+		        GroovyServices services = new GroovyServices(new CompilationUnitFactory());
+		        services.setWorkspaceRoot(workspaceRoot);
+		        services.connect(new LanguageClient() {
+
+					@Override
+					public void telemetryEvent(Object object) {
+
+					}
+
+					@Override
+					public CompletableFuture<MessageActionItem> showMessageRequest(ShowMessageRequestParams requestParams) {
+						return null;
+					}
+
+					@Override
+					public void showMessage(MessageParams messageParams) {
+
+					}
+
+					@Override
+					public void publishDiagnostics(PublishDiagnosticsParams diagnostics) {
+
+					}
+
+					@Override
+					public void logMessage(MessageParams message) {
+
+					}
+				});
+		        
+		        Path filePath = workspaceRoot.resolve("Completion.groovy");
+				String uri = filePath.toUri().toString();
+				TextDocumentItem textDocumentItem = new TextDocumentItem(uri, LANGUAGE_GROOVY, 3, document.get());
+				services.didOpen(new DidOpenTextDocumentParams(textDocumentItem));
+				TextDocumentIdentifier textDocument = new TextDocumentIdentifier(uri);
+
+				String substring =  document.get().substring(0,offset);
+				long line = substring.chars().filter(ch -> ch == '\n').count();
+				int column = 0;
+				
+				if (line == 0) {
+					column = offset;
+				} else {
+					String[] allLine = substring.split("\n");
+					int length = allLine.length;
+					String lastLine = allLine[length-1];
+					column = lastLine.length();
+				}
+				
+				Position position = new Position((int) line, column);
+				Either<List<CompletionItem>, CompletionList> result = null;
+				
+				try {
+					result = services.completion(new CompletionParams(textDocument, position)).get();
+				} catch (InterruptedException | ExecutionException e) {
+					e.printStackTrace();
```

Comment:
Please log the `Exception` using a `Logger` like `LOGGER.error("Error on getting the completion service result.", e);`